src/main/java/de/tum/in/www1/artemis/service/connectors/gitlab/GitLabService.java

```diff
@@ -342,7 +342,7 @@ public void createProjectForExercise(ProgrammingExercise programmingExercise) th
         for (final var editor : editors) {
             try {
                 final var userId = gitLabUserManagementService.getUserId(editor.getLogin());
-                gitLabUserManagementService.addUserToGroups(userId, List.of(programmingExercise), DEVELOPER);
+                gitLabUserManagementService.addUserToGroupsOfExercises(userId, List.of(programmingExercise), DEVELOPER);
```

Comment:
Did you have checked that a developer can push to the main branch? In some cases the main branch is protected and I think a developer may not be allowed to push. This should be checked :)

Reply:
You're right, developers cannot push to protected branches. According to the Gitlab docs, only people with `MAINTAINER` and `OWNER` roles are allowed to do that. 

This is might also be the reason behind #3413. According to the [Gitlab permissions docs](https://docs.gitlab.com/ee/user/permissions.html), I think that the new editor role needs to have the `MAINTAINER` role where as the instructor should be assigned the `OWNER` role. What do you think @florian-glombik

Reply:
Yes, the editor should be able to change the code in the repository through git as well.
But he should not be able to delete something or do anything similar. So Maintainer should be the correct role for the editor

Reply:
@dfuchss I changed the roles which should fix #3413

Reply:
That would be my guess too :)